File: SpringBoot/src/main/java/com/group3s2i/springboot/Model/OrderCustomer.java
package com.group3s2i.springboot.Model;


import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@ToString
@Entity
@Table(name="orders_customer")
public class OrderCustomer implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "order_number")
    private String orderNumber;

    @Column(name = "status")
    private String status;

    @Column(name = "sent_media")
    private String sentMedia;

    @Column(name = "urgent")
    private String urgent;

    @Column(name = "created_date")
    private LocalDateTime createdDate;


    @OneToMany(
            mappedBy = "orderCustomer",
            cascade = CascadeType.ALL,
            orphanRemoval = true
    )
    @JsonIgnoreProperties
    @ToString.Exclude
    @JsonManagedReference(value = "m-d")
    private List<OrderCustomerItem> orderCustomerItems;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name="customer_id", referencedColumnName = "id")
    @JsonBackReference(value = "o-s")
    @ToString.Exclude
    private Customer customer;

    public OrderCustomer() {
    }


}

File: ecommerce/lib/models/OrderCustomer.dart
import 'OrderCustomerItem.dart';
import 'User.dart';

class OrderCustomer {
  OrderCustomer({
    this.id,
    this.orderNumber,
    this.status,
    this.sentMedia,
    this.urgent,
    this.createdDate,
    this.cancelledDate,
    this.orderDate,
    this.orderCustomerItems,
    this.user
  });

  int id;
  String orderNumber;
  String status;
  String sentMedia;
  String urgent;
  DateTime createdDate;
  DateTime cancelledDate;
  DateTime orderDate;
  List<OrderCustomerItem> orderCustomerItems;
  User user;

  factory OrderCustomer.fromJson(Map<String, dynamic> json) => OrderCustomer(
    id: json["id"],
    orderNumber: json["orderNumber"],
    status: json["status"],
    sentMedia: json["sentMedia"],
    urgent: json["urgent"],
    createdDate: DateTime.parse(json["createdDate"]),
    cancelledDate: json["cancelledDate"],
    orderDate: json["orderDate"],
    orderCustomerItems: List<OrderCustomerItem>.from(json["orderCustomerItems"].map((x) => OrderCustomerItem.fromJson(x))),
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "orderNumber": orderNumber,
    "status": status,
    "sentMedia": sentMedia,
    "urgent": urgent,
    "createdDate": createdDate.toIso8601String(),
    "cancelledDate": cancelledDate,
    "orderDate": orderDate,
    "orderCustomerItems": List<dynamic>.from(orderCustomerItems.map((x) => x.toJson())),
  };
}
